src/data_channel/mod.rs

```diff
@@ -305,18 +305,31 @@ impl RTCDataChannel {
                 _ = notify_rx.notified() => break,
                 result = data_channel.read_data_channel(&mut buffer) => {
                     match result{
+                        Ok((0, _)) =>
```

Comment:
Hmmmm 🤔 This gets me thinking: is there a convention in the project to return EOF as an error that we are violating by changing it to `Ok(0)`?